bin/specextract

```diff
@@ -828,860 +753,1086 @@ def clip_wmap(wmapfile,threshold,tmpdir):
+                         comment="galactic HI column density") 
+        except NameError:
+            pass
+            
+        try:
+            # Galactic neutral hydrogen column density at the source position using
+            # the Bell Labs survey, via COLDEN.
 
-    arfcorr()
+            edit_headers(verbose,phafile,"Bell_nH",float(f"{bell_nh:.6}"),unit="10**22 cm**-2",
+                         comment="galactic HI column density")
+        except NameError:
+            pass
 
-    # add arfcorr history to file, since it doesn't automatically
-    param_arfcorr = {}
-    for acpar in arfcorr._parnames:
-        param_arfcorr[acpar] = arfcorr._get_param_value(acpar)
 
-    add_tool_history(carffile,"arfcorr",param_arfcorr)
-    
-    return carffile
+        ############################################################
+        #
+        # add AREASCAL value from 'blanksky' BKGSCAL value, do not
+        # change if using 'blanksky_sample'
+        #
+        ############################################################
+        
+        if srcbkg == "bkg":
+            dmhistory.punlearn()
+            dmhistory.infile = fullfile
 
+            try:
+                dmhistory.tool = "blanksky"
+                bsky_status = dmhistory()
+            except OSError:
+                bsky_status = None
 
+            try:
+                dmhistory.tool = "blanksky_sample"
+                bskysamp_status = dmhistory()
+            except OSError:
+                bskysamp_status = None
+                
+            if all([bsky_status is not None, bskysamp_status is None]):
+                src_cr = pcr.read_file(f"{filename}[#row=0]")
+                ccdid = src_cr.get_column("CCD_ID").values[0]
+                del(src_cr)
 
-##########################################################################
-# Usage:
-#   group_spectrum(ptype, full_outroot, val, spec, gtype,
-#		    clobber, verbose, phafile)
-#
-# Aim:
-#   Optionally group output spectrum.
-#
-##########################################################################
+                bkgscale = fileio.get_keys_from_file(fullfile)[f"BKGSCAL{ccdid}"]
 
-def group_spectrum(ptype, full_outroot, val, spec, gtype,
-                    clobber, verbose, phafile):
+                edit_headers(verbose, phafile, "AREASCAL", bkgscale, comment=f"blanksky BKGSCAL{ccdid} scaling factor from the 'blanksky' script, set to 1.0 if subtracting background")
 
-    # grouped spectrum name
-    grpout = f"{full_outroot}_grp.{ptype.lower()}"
+                v1(f"{phafile} AREASCAL keyword set to the 'blanksky' BKGSCAL{ccdid} value for automated background spectrum scaling while spectral modeling.  Set AREASCAL to 1.0 if 'blanksky' spectrum is being used for background subtraction.")
 
-    dmgroup.punlearn()
+    try:
+        outreg.close()
+    except NameError:
+        pass
+                
+    try:
+        return pardict
+    except NameError:
+        pass
```

Comment:
This looks really confusing (especially as it's the last line in the function). Why is it okay to ignore a `NameError` and what's meant to happen when it happens?